include/dlaf/matrix/hdf5.h

```diff
@@ -15,6 +15,7 @@
 #include <complex>
 #include <cstdint>
 #include <string>
+#include <string_view>
```

Comment:
```suggestion
#include <string_view>
#include <typeinfo>
```
because of (https://en.cppreference.com/w/cpp/language/typeid):
> The header [<typeinfo>](https://en.cppreference.com/w/cpp/header/typeinfo) must be [included](https://en.cppreference.com/w/cpp/standard_library#Including_headers) or [imported](https://en.cppreference.com/w/cpp/standard_library#Importing_headers)(since C++20) or a [standard library module](https://en.cppreference.com/w/cpp/standard_library#Importing_modules) imported(since C++23) before using typeid, otherwise the program is ill-formed.

Reply:
TIL, thanks!